### PR TITLE
GUI overhaul: store window maximization status, resizable resultDisplay, window height that scales by system display resolution

### LINKED ISSUE #118: Prevent text from getting cut off in UI
Allow text to wrap-around horizontally and increase the vertical height of the window to prevent text from getting cut off.

Current UI:
image

### LINKED ISSUE #170: Window elements cut off on lower resolutions
At 1280x720, 150% scaling:
WindowsTerminal_PtfH2Lb1k9

### PR DESCRIPTION
**Pull request by aureliony**
Closes #170 
Closes #118

### CODE REVIEW

**Comment by aureliony** [comment]
Removed the maximization of window by default. 1280x720@150% scaling can run properly if the default height is set to 450 pixels. However, 1080p+ user experience will not be optimal.  The current app looks like this:

image

@bryanyee33 Would implementing the height scaling by screen resolution be better?

File: src/main/java/seedu/address/ui/MainWindow.java [hunk 1]
```diff
@@ -143,6 +144,11 @@ private void fillInnerParts() {
     private void setWindowDefaultSize(GuiSettings guiSettings) {
         primaryStage.setHeight(guiSettings.getWindowHeight());
         primaryStage.setWidth(guiSettings.getWindowWidth());
+
+        if (guiSettings.getIsMaximized()) {
+            primaryStage.setMaximized(true);
```
Comment (aureliony) [inline src/main/java/seedu/address/ui/MainWindow.java:149]: Would it be better to replace this with primaryStage.setMaximized(guiSettings.getIsMaximized())?

File: src/test/data/JsonUserPrefsStorageTest/ExtraValuesUserPref.json [hunk 2]
```diff
@@ -7,7 +7,8 @@
       "x" : 300,
       "y" : 100,
       "z" : 99
-    }
+    },
+    "isMaximized" : false
   },
```
Comment (aureliony) [inline src/test/data/JsonUserPrefsStorageTest/ExtraValuesUserPref.json:12]: Should this be updated with the divider position?

File: src/test/data/JsonUserPrefsStorageTest/TypicalUserPref.json [hunk 3]
```diff
@@ -5,7 +5,8 @@
     "windowCoordinates" : {
       "x" : 300,
       "y" : 100
-    }
+    },
+    "isMaximized" : false
```
Comment (aureliony) [inline src/test/data/JsonUserPrefsStorageTest/TypicalUserPref.json:9]: Should this be updated with the divider position?

Comment (bryanyee33) [reply src/test/data/JsonUserPrefsStorageTest/TypicalUserPref.json]: Yes, it has been updated

File: src/test/data/JsonUserPrefsStorageTest/ExtraValuesUserPref.json (hunk 2 above)
Comment (bryanyee33) [reply src/test/data/JsonUserPrefsStorageTest/ExtraValuesUserPref.json:13]: Yes, it has been updated

File: src/main/java/seedu/address/ui/MainWindow.java (hunk 1 above)
Comment (bryanyee33) [reply src/main/java/seedu/address/ui/MainWindow.java]: Yes, it has been changed

File: src/main/java/seedu/address/commons/core/GuiSettings.java [hunk 4]
```diff
@@ -14,27 +15,48 @@ public class GuiSettings implements Serializable {
...
-     * Constructs a {@code GuiSettings} with the default height, width and position.
+     * Constructs a {@code GuiSettings} with the default parameters.
      */
     public GuiSettings() {
-        windowWidth = DEFAULT_WIDTH;
-        windowHeight = DEFAULT_HEIGHT;
+        try {
+            Rectangle2D primaryScreenBounds = Screen.getPrimary().getVisualBounds();
+            windowWidth = primaryScreenBounds.getWidth() * 0.5;
+            windowHeight = primaryScreenBounds.getHeight() * 0.9;
```
Comment (aureliony) [inline src/main/java/seedu/address/commons/core/GuiSettings.java:33]: Should these magic numbers be replaced with private static variables?

File: src/main/java/seedu/address/commons/core/GuiSettings.java [hunk 5]
```diff
@@ -14,27 +15,48 @@ public class GuiSettings implements Serializable {
...
+        } catch (ExceptionInInitializerError | NoClassDefFoundError e) { // No screen found
+            windowWidth = DEFAULT_WIDTH;
+            windowHeight = DEFAULT_HEIGHT;
+        }
         windowCoordinates = null; // null represent no coordinates
+        isMaximized = false;
```
Comment (aureliony) [inline src/main/java/seedu/address/commons/core/GuiSettings.java:39]: Should these default values also be replaced with private static variables?

Comment (bryanyee33) [reply src/main/java/seedu/address/commons/core/GuiSettings.java]: I have put them as default values when initializing

File: src/main/java/seedu/address/commons/core/GuiSettings.java (hunk 4 above)
Comment (bryanyee33) [reply src/main/java/seedu/address/commons/core/GuiSettings.java]: Yes, I have changed it

File: src/main/java/seedu/address/commons/core/GuiSettings.java [hunk 6]
```diff
@@ -63,21 +93,14 @@ public boolean equals(Object other) {
         GuiSettings otherGuiSettings = (GuiSettings) other;
         return windowWidth == otherGuiSettings.windowWidth
                 && windowHeight == otherGuiSettings.windowHeight
-                && Objects.equals(windowCoordinates, otherGuiSettings.windowCoordinates);
+                && Objects.equals(windowCoordinates, otherGuiSettings.windowCoordinates)
```
Comment (yisiox) [inline src/main/java/seedu/address/commons/core/GuiSettings.java:96]: Is there a reason for using `Objects.equals` vs `A.equals(B)`?

**Review by yisiox — APPROVED** [review]
LGTM, I don't see anything that may be an issue but see comment still.

File: src/main/java/seedu/address/commons/core/GuiSettings.java (hunk 6 above)
Comment (yisiox) [reply src/main/java/seedu/address/commons/core/GuiSettings.java:96]: Update: it's null safe, nice.

**Pull request merged**